### DISCUSSION TITLE
rust-course/first-try/intro

### DISCUSSION THREAD

**Discussion by sunface** [sunface/rust-course#607] [章节评论区]
https://course.rs/first-try/intro.html

**Comment by sharkLoc** [comment]
一路看到泛型和特征，这两者结合起来后看着有点懵，先跳过，后面再回来补。

**Comment by sunface** [comment]
是否因为书本的内容上存在一些讲解不清楚的地方？

**Comment by lavarabbit** [comment]
> @sharkLoc
> 一路看到泛型和特征，这两者结合起来后看着有点懵，先跳过，后面再回来补。

可以把Generics对应c++的template，Trait对应abstract class/interface

**Reply by mrhanice** [reply]
学过scala,看到trait竟觉得好熟悉

**Reply by GLORYAIMER** [reply]
我也是，我学了那么多语言c，c++，py，java，js，感觉都是依托于指针结构体的，像这门语言我也学了六七天，也还是不太能get到泛型和特征的点

**Comment by sharkLoc** [comment]
> @sunface
> 是否因为书本的内容上存在一些讲解不清楚的地方？
那倒不是，讲的挺好的

**Comment by kozyan** [comment]
讲得很详细，容易看得明白。

**Comment by ClearPlume** [comment]
> @lavarabbit
> > @sharkLoc
> > 一路看到泛型和特征，这两者结合起来后看着有点懵，先跳过，后面再回来补。
> 
> 可以把Generics对应c++的template，Trait对应abstract class/interface

最开始的时候可以这样理解方便切入，但细微处的区别还是挺大...甚至夸大一点，都能说基本不相关，只有最表面一层比较相似了

**Comment by mxismean** [comment]
👍🏻 学习

**Comment by ghost** [comment]
马上入门！第一章！

**Comment by axzed** [comment]
mark~2022-10-30

**Reply by Thyiad** [reply]
学成没

**Comment by inkwise** [comment]
学习打卡

**Comment by showyou2023** [comment]
建议把官方教程中的实例项目:一个I/O项目引入本书，学习基本知识后，动手做这个项目是非常有意义的。

**Comment by wsvaio** [comment]
学无止境

**Comment by rcx100** [comment]
开始学习 感谢

**Comment by yjshi2015** [comment]
牛刀牛刀，学习Rust的第1天，心情是愉悦的~

**Comment by liuclub** [comment]
mark

**Comment by yanqing-j** [comment]
第一天学习，脸上的表情是这样的充满着快乐的

**Comment by mxhjhbrtggc** [comment]
泛型和特征太难了

**Comment by Pup4tor** [comment]
学习打卡：）

**Comment by Sifotd** [comment]
希望这次能成功入门

**Comment by gudeyu** [comment]
开始学Rust，看看和 C++ 差别多大

**Comment by GLORYAIMER** [comment]
mid-mark:2025.8.1 又是被人类智慧征服的一天 Orz我是F5 Orz我是F5 Orz我是F5 Orz我是F5 Orz我是F5 Orz我是F5 Orz我是F5 Orz我是F5 Orz我是F5 Orz我是F5 Orz我是F5

**Reply by GLORYAIMER** [reply]
[
<img width="287" height="263" alt="image" src="https://github.com/user-attachments/assets/0dfd7257-5daf-4bf6-9f63-471a8233d8dd" />](url)
